### DISCUSSION TITLE
Yolo11_pose cannot separate person when overlap happened

### DISCUSSION THREAD

**Discussion by Ying5774** [ultralytics/ultralytics#21777] [General]
As showed in the following image, I want to track the woman with full face showed in image, however yolo lost tracking when another person showed up. It gave me a single one bounding box with two people, then begin to change tracking target from the woman to the other person from this point.
<img width="373" height="552" alt="image" src="https://github.com/user-attachments/assets/af64ab70-fe40-4a9f-b3da-5744a492cb33" />
Is there any other better way to separate the bounding boxes between the target person and other person?
Thanks!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Ying5774, thank you for your interest in Ultralytics 🚀! This is an automated response — an Ultralytics engineer will also review your discussion and assist soon.

For quick starts and best practices, please visit the [Docs](https://docs.ultralytics.com/) where you’ll find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions are already addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom tracking/pose ❓ Question, please include:
-Exact Ultralytics version and OS/Device (CPU/GPU)
-Full command or Python snippet used (model, task=pose, tracker settings, conf/iou/imgsz, etc.)
-Tracker config file if used (e.g., bytetrack/ocsort)
-A short video clip or a few consecutive frames showing the overlap case
-Any logs/warnings and a brief description of what you expected vs. observed
-And verify you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) if you trained or fine-tuned a model

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

-**Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
-**Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
-**Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
-**Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Thanks for the clear example—this is typically the detector merging two people under heavy overlap. Try: 1) making NMS less aggressive and increasing resolution/model size, and 2) enabling ReID in the tracker for better ID stability through occlusion.

Example:
```bash
yolo track model=yolo11m-pose.pt source=path/to/video.mp4 imgsz=1280 iou=0.85 conf=0.25 tracker=botsort.yaml
```

To use ReID, copy `ultralytics/cfg/trackers/botsort.yaml` and set:
```yaml
with_reid: True
model: auto
track_buffer: 30
match_thresh: 0.8
proximity_thresh: 0.5
appearance_thresh: 0.25
```

You can also compare `tracker=bytetrack.yaml` in crowded scenes. See the tracking guide and ReID notes in the [Multi-Object Tracking docs](https://docs.ultralytics.com/modes/track/).

**Comment by Ying5774** [comment]
Thank you Glenn! Sorry for the late reply. I would enable ReID but can increase neither resolution nor model size at this moment, because it's a real-time APP and running on a Robot. Has to be yolo11n to gain fast response.

**Reply by glenn-jocher** [reply]
Totally fine to stay on YOLO11n—optimize the detector/tracker instead of size. Try higher NMS IoU, slightly lower `conf`, and BoT-SORT ReID with native features to keep IDs through overlap:

```bash
yolo track model=yolo11n-pose.pt source=path/to/video.mp4 iou=0.85 conf=0.20 tracker=my_botsort.yaml
```

```yaml
# my_botsort.yaml
with_reid: True
model: auto            # uses YOLO features; minimal overhead
track_buffer: 60       # keep lost IDs longer through occlusion
match_thresh: 0.8
proximity_thresh: 0.5
appearance_thresh: 0.25
fuse_score: True
gmc_method: sparseOptFlow  # helpful if the camera moves
```

If ReID cost is still too high, compare `tracker=bytetrack.yaml` (often strong in crowds). Config details and ReID notes are in the Multi‑Object Tracking docs under [Tracker arguments](https://docs.ultralytics.com/modes/track/#tracker-arguments) and [Enabling ReID](https://docs.ultralytics.com/modes/track/#enabling-re-identification-reid).
